user_tools/src/spark_rapids_tools/tools/qualx/preprocess.py

```diff
@@ -448,6 +452,10 @@ def combine_tables(table_name: str) -> pd.DataFrame:
                      fallback_reason=f'Empty feature tables found after preprocessing: {empty_tables_str}')
         return pd.DataFrame()
 
+    if get_label() == 'duration_sum':
+        # override appDuration with sum(duration_sum) across all stages
+        app_tbl['appDuration'] = job_stage_agg_tbl['duration_sum'].astype(float).sum()
```

Comment:
Does duration_sum only include sql ids?   I guess no way to get duration for sum for non sql parts?

Reply:
yes, although I'm seeing some weirdnesses w/ this number so taking another look.

Reply:
Fixed this code to correctly aggregate by appId, but the duration_sum is still only derived from whatever is available in the job_stage_agg_tbl.